File: app/src/main/java/com/gb/rating/models/repository/UnverifiedRatingRepository.java
package com.gb.rating.models.repository;

import com.gb.rating.models.UnverifiedRating;
import java.util.List;
import io.reactivex.Maybe;
import io.reactivex.annotations.NonNull;

public interface UnverifiedRatingRepository {

    @NonNull io.reactivex.Completable writeUnverifiedRating(@NonNull UnverifiedRating value, @NonNull String iMEI);
    @NonNull Maybe<List<UnverifiedRating>> retrieveUnverifiedRatingList(@NonNull String iMEI);
}

File: app/src/main/java/com/gb/rating/models/repository/UnverifiedRatingRepository.kt
package com.gb.rating.models.repository

import com.gb.rating.models.UnverifiedRating
import io.reactivex.Maybe
import io.reactivex.annotations.NonNull

interface UnverifiedRatingRepository {

    @NonNull
    fun writeUnverifiedRating(@NonNull value: UnverifiedRating, @NonNull iMEI: String): io.reactivex.Completable

    @NonNull
    fun retrieveUnverifiedRatingList(@NonNull iMEI: String): Maybe<List<UnverifiedRating>>
}

